src/Serialization/main.jl

```diff
@@ -146,6 +146,28 @@ end
 params(tp::TypeParams) = tp.params
 type(tp::TypeParams) = tp.type
 
+@doc """
+    type_params(obj::T) where T
+
+Returns the parameters needed for the serialization of `T` as a `TypeParams`,
+
+# Example
+```
+julia> R1, (x, y) = graded_polynomial_ring(QQ, [:x, :y])
+(Graded multivariate polynomial ring in 2 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y])
+
+julia> save("/path/to/file.mrdi", [x^2, x - y])
+
+julia> R2, (s, t) = graded_polynomial_ring(QQ, [:s, :t])
+(Graded multivariate polynomial ring in 2 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[s, t])
+
+julia> load("/path/to/file.mrdi"; params=Oscar.type_params([s, t]))
```

Comment:
Can't we turn this into an actual docstring which gets tested and updated if outputs change?
````suggestion
```jldoctest
julia> R1, (x, y) = graded_polynomial_ring(QQ, [:x, :y])
(Graded multivariate polynomial ring in 2 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[x, y])

julia> save("example.mrdi", [x^2, x - y])

julia> R2, (s, t) = graded_polynomial_ring(QQ, [:s, :t])
(Graded multivariate polynomial ring in 2 variables over QQ, MPolyDecRingElem{QQFieldElem, QQMPolyRingElem}[s, t])

julia> load("example.mrdi"; params=Oscar.type_params([s, t]))
````

Reply:
this would then also need a `teardown` thingy to delete the file afterwards

Reply:
see latest commits.